graph/adjacency/impl/ThingAdjacencyImpl.java

```diff
@@ -304,10 +304,10 @@ private void put(Encoding.Edge.Thing encoding, ThingEdgeImpl.Buffered edge, IID[
                     if (existingEdge == null) {
                         if (isOut()) owner.graph().edgeCreated(edge); // only record creation in one direction
                         return edge;
-                    } else if (existingEdge.isInferred() && !edge.isInferred()) {
-                        existingEdge.isInferred(true);
+                    } else {
+                        assert existingEdge.isInferred() == edge.isInferred();
```

Comment:
for edges, we also check that the existing edge has the expected inferred status